중고물품 변경 및 삭제

junshock5 opened the pull request:
accountID를 통한 본인 중고거래 물품 Select 기능 추가

junshock5 committed:
accountID를 통한 본인 중고거래 물품 Select 기능 추가

f-lab-dev commented on src/main/java/com/market/server/controller/ProductController.java:
```diff
@@ -1,23 +1,37 @@
 @Log4j2
 public class ProductController {
+
+    @Autowired
```
여기서 field injection과 constructor injection을 혼용하고 있는데 한가지 방식만 사용하시는게 좋습니다~

junshock5 replied:
- Constructor Injection를 통해 얻는 장점
1. Component를 변경이 불가능한 immutable 상태로 선언이 가능합니다.
2. 더 나아가 생성자의 Parameter를 통해 의존관계를 한눈에 파악하고 Refactoring의 필요성을 얻을 수 있습니다.
3. Component간 순환 참조를 하고 있다면 BeanCurrentlyInCreationException이 발생해서 순환 의존성을 쉽게 알 수 있습니다.
4. 특정 DI Container에 의존하지 않으며 쉽게 단위 테스트도 가능하고 필요하다면 다른 DI Container로 전환이 가능합니다.
5. DI Container가 Component를 주입을 못한 경우 실제 Method가 호출되기 이전인 생성자에서 Handling 할 수 있습니다.

f-lab-dev commented on src/main/java/com/market/server/controller/ProductController.java:
```diff
@@ -29,4 +43,25 @@ public void registerProduct(@RequestBody ProductDTO productDTO,
+    /**
+     * 본인 중고물품 검색 메서드.
+     */
+    @GetMapping("selectMyProducts")
```
이미 `GET` 메소드에는 `select`라는 의미가 들어있습니다. 
URL로 기능을 표현하시는거에 대해 익숙하지 않으신 것 같은데 이해하면서 문서를 봐주세요~

junshock5 replied:
변경하겠습니다.

f-lab-dev commented on src/main/java/com/market/server/service/Impl/ProductServiceImpl.java:
```diff
@@ -39,4 +37,10 @@ public void register(String id, ProductDTO productDTO) {
     }
+
+    @Override
+    public List<ProductDTO> selectMyProducts(int accountId) {
```
`select`는 DB관점의 용어입니다. 그러면 비즈니스로직을 정의하는 `service`에서는 어떤 단어를 사용해야 DB에 종속적이지 않은 단어를 사용할 수 있을까요?

junshock5 replied:
getMyProducts 으로 변경 하겠습니다.

f-lab-dev commented on src/main/resources/mybatis/mapper/productMapper.xml:
```diff
@@ -31,4 +31,20 @@
 
+    <select id="selectMyProducts" resultType="com.market.server.dto.ProductDTO">
+        SELECT `id`,
+        `price`,
```
여기도 필드는 들여쓰기를 해주시는게 좋아보입니다~

junshock5 replied:
feature/13에서 변경하였습니다.

junshock5 committed:
ProductController에서 constructor injection 으로 변경, API 메서드명 변경

junshock5 committed:
selectMyProducts -> getMyProducts 메소드 명 변경에따른 호출 부분 변경

junshock5 committed:
중고물품 수정,삭제, product writerId-> accountId필드명 변경

f-lab-dev commented on src/main/java/com/market/server/controller/ProductController.java:
```diff
@@ -53,7 +54,42 @@ public ProductResponse myProductInfo(HttpSession session) {
+    /**
+     * 본인 중고물품 수정 메서드.
+     */
+    @PatchMapping("{productId}/update")
```
`Patch`라는 단어에 update라는 뜻이 담겨있습니다.
이전에 링크달아드린 API 설계 가이드 문서를 꼼꼼하게 봐주세요~

리뷰를 적용할때는 단순 리뷰 내용 적용으로 끝나면 안됍니다. 꼭 리뷰를 달아드린 이유를 파악하시고 추후의 코드에 적용하도록 해주세요

junshock5 replied:
commit id: d09f4d0 여기서 
REST API 참고사이트: https://spoqa.github.io/2012/02/27/rest-introduction.html
참고하여 업데이트 하였습니다.

f-lab-dev commented on src/main/java/com/market/server/controller/ProductController.java:
```diff
@@ -53,7 +54,42 @@ public ProductResponse myProductInfo(HttpSession session) {
+        productDTO.setIstrade(productRequest.isTrade());
+        productDTO.setUpdatetime(new Date());
+        productDTO.setDeliveryprice(productRequest.getDeliveryprice());
+        productDTO.setDibcount(productRequest.getDibcount());
```
이런 긴 코드는 팩토리 메소드로 제공하거나 생성자를 이용해주세요~ 생성자로 만들 수 있음에도 굳이 setter를 사용하면 `immutable`을 깨지 않아도 될 상황에 깨게 될 수 있습니다

junshock5 replied:
eb5ab11 적용하였습니다.

f-lab-dev commented on src/main/resources/mybatis/mapper/productMapper.xml:
```diff
@@ -44,7 +44,27 @@
+            deliveryprice = #{deliveryprice},
+            dibcount = #{dibcount}
+        WHERE accountId = #{accountId}
+        AND id = #{id}
```
여기는 왜 조건이 2개인가요? id가 있다면 굳이 조건을 2개 달아주지 않아도 될 것 같네요~

junshock5 replied:
bf6741f 적용하였습니다.

junshock5 committed:
ProductController의
본인 중고물품 수정, 삭제 url패턴을
REST API URL 패턴 규칙으로 변경
참고 사이트: https://spoqa.github.io/2012/02/27/rest-introduction.html

junshock5 committed:
중고 물품 수정 ProductDTO 생성시 객체 immutable 를 유지하기 위한 생성자 생성으로 변경

junshock5 committed:
updateProducts sql문의 where절에 불필요한 accountId 제거

junshock5 committed:
Merge branch 'develop' into feature/17
